app/common/update.py

```diff
@@ -231,3 +258,90 @@ def merge_local_db():
+        
+
+def download_dat_files():
+    """
+    Verifies the user's DQX install location and prompts
+    them to locate it if not found. Uses this location to
+    download the latest data files from the dqxclarity repo.
+    """
+    config = load_user_config()
+    dat0_file = "data00000000.win32.dat0"
+    idx_file = "data00000000.win32.idx"
+    
+    # If user_settings.ini doesn't have the install directory
+    # set, let's set it now
+    
+    if not config["config"]["installdirectory"]:
+    
+        # First, let's check the default location and see if it's installed there
+        default_path = 'C:/Program Files (x86)/SquareEnix/DRAGON QUEST X/Game/Content/Data'
+        
+        if os.path.exists(default_path):
+            update_user_config('config', 'installdirectory', default_path)
+        else:
+            # Doesn't exist. Let's prompt the user for it
+            warning_message(
+            title="[dqxclarity] Couldn't Find Directory",
+            message="Could not find DQX directory. Please select\nthe Data folder located in your\nDQX install location."
+            )    
+            
+            # Prompt the user to select their DQX data directory, and check
+            # if the dat0 file exists in this path. If it does, it *should* be legit.
+            # Keep prompting the user until the path is valid
+            while True:
+                dqx_path = askdirectory()
+                dat0_path = dqx_path + "/" + dat0_file
+                
+                if os.path.isfile(dat0_path)
+                    update_user_config('config', 'installdirectory', dqx_path)
+                    break
+                else:
+                    warning_message(
+                        title="[dqxclarity] Invalid Directory",
+                        message="The path you provided is not a valid DQX path.\nPlease select the Data folder located\nin your DQX install location."
+                    )
+    
+    # We have an installation path, so let's check it
+    
+    if config["config"]["installdirectory"]:
+        
+        dqx_path = config["config"]["installdirectory"]
+        dat0_path = dqx_path + "/" + dat0_file
+        
+        # If the dat0 file exists here, it should be legit
+        # and we can move on
+        
+        if os.path.isfile(dat0_path):
+            # First, let's check if an idx file backup already exists
+            idx_path = dqx_path + "/" + idx_file + ".bak"
+            
+            # If it doesn't, let's back it up first with a copy
+            if not os.path.isfile(idx_path):
+                shutil.copy(dqx_path + "/" + idx_file, dqx_path + "/" + idx_file + ".bak")
+            
+            # Now let's download the data files
+            try:
+                logger.info("Downloading DAT1 and IDX files.")
+                request = requests.get(GITHUB_CLARITY_DAT1_URL, timeout=15)
+                if request.status_code == 200:

```

Comment:
We will need to handle what happens when we don't get a status_code of 200 as otherwise, it will just skip to the next file and eventually write "success". Not getting a 200 won't trigger an exception, so we will want to handle it.

I would make both GET calls first and make sure they both come back successfully before we write so we don't only partially write the changes to the game files.